Lib/unittest/runner.py

```diff
@@ -184,8 +184,11 @@ def run(self, test):
         if hasattr(result, 'separator2'):
             self.stream.writeln(result.separator2)
         run = result.testsRun
-        self.stream.writeln("Ran %d test%s in %.3fs" %
-                            (run, run != 1 and "s" or "", timeTaken))
+        if run > 0 or len(result.skipped) > 0:
```

Comment:
I'd abstract the  `run > 0 or len(result.skipped) `  calculation as a method on `TestResult`, then you won't have it repeated twice.  And you can give it a name indicating what it means.